app/controllers/catalog_controller.rb

```diff
@@ -929,23 +924,21 @@ def atomic_req_submit
 
     # Check the validity of the entry points
     %i(fqname reconfigure_fqname retire_fqname).each do |fqname|
-      if @edit[:new][fqname].present? &&
-         MiqAeClass.find_homonymic_instances_across_domains(current_user, @edit[:new][fqname]).empty?
-        case fqname
-        when :fqname
-          add_flash('Please correct invalid Provisioning Entry Point prior to saving', :error)
-        when :reconfigure_fqname
-          add_flash('Please correct invalid Reconfigure Entry Point prior to saving', :error)
-        when :retire_fqname
-          add_flash('Please correct invalid Retirement Entry Point prior to saving', :error)
-        end
+      next unless @edit[:new][fqname].present? && MiqAeClass.find_homonymic_instances_across_domains(current_user, @edit[:new][fqname]).empty?
```

Comment:
Maybe better as 

```
next if @edit[:new][fqname].blank?
next unless MiqAeClass.find_homonymic_instances_across_domains(current_user, @edit[:new][fqname]).empty?
```

(or `next if @edit[:new][fqname].blank? || !MiqAeClass.find_homonymic_instances_across_domains(current_user, @edit[:new][fqname]).empty?` ?)

(The combination of `unless` and complex logic with `&&` is what I don't like.)

Reply:
Fixed